workers/queue.ts

```diff
@@ -13,6 +13,36 @@ const maxWorkerTimeSeconds = 120;
 const maxWorkerThreads = 5;
 let currentWorkerThreads = 0;
 
+/**
+ * Create a worker that can execute TypeScript modules using ts-node/register.
+ *
+ * @param file Path to worker file
+ * @param workerOptions Worker options
+ */
+const workerTs = (file: string, workerOptions: WorkerOptions) => {
+	workerOptions.eval = true;
+
+	if (!workerOptions.workerData) {
+		workerOptions.workerData = {};
+	}
+
+	// Inject the file name into the worker.
+	workerOptions.workerData.__filename = file;
+
+	// Create a worker that evals a script which registers ts-node prior to
+	// requiring the entrypoint.
+	return new Worker(
+		`
+					const wk = require('worker_threads');
+					require('ts-node').register();
+					let file = wk.workerData.__filename;
+					delete wk.workerData.__filename;
+					require(file);
+			`,
+		workerOptions,
+	);
+};
```

Comment:
**Important change**  — workers are now spawned using `eval` which allows us to inject a call to `require('ts-node').register()` prior to importing the task module.